src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1016,31 +1016,29 @@ private function registerRouterConfiguration(array $config, ContainerBuilder $co
                 ->replaceArgument(0, $config['default_uri']);
         }
 
-        if (\PHP_VERSION_ID >= 80000 || $this->annotationsConfigEnabled) {
```

Comment:
We should turn that into an early return statement in 5.2 to reduce merge conflict!

Reply:
#41345

Reply:
Please rebase.